src/bundles/SellerRegistration/components/DocumentsForm/DocumentsForm.js

```diff
@@ -0,0 +1,178 @@
+    { 'label': 'Public Liability Insurance', 'id': 'liability' },
+    { 'label': 'Workers Compensation Insurance', 'id': 'workers' },
+    { 'label': 'Signed copy of Deed of Standing Offer', 'id': 'deed' },
+  ]
+
+  onUpload(id, e) {
+    e.preventDefault();
+    const { dispatch, model, action, csrf_token } = this.props;
+    const file = this.state[id].file;
+
+    this.setState({
+      [id]: Object.assign({}, this.state[id], {'uploading': true, 'file': void 0}),
+      errors: Object.assign({}, this.state.errors, {[id]: void 0})
+    })
+
+    dispatch(actions.omit(`${model}.documents`, id))
+
+    let data = new FormData()
+    data.append(id, file)
+    fetch(`${action}/${id}`, {
+      method: 'POST',
+      body: data,
+      credentials: 'same-origin',
+      headers: {
+        'X-CSRFToken': csrf_token
+      }
+    })
+    .then(response => {
+      if (response.status >= 200 && response.status < 300) {
+        return response
+      } else {
+        var error = new Error(response.statusText)
+        error.response = response
+        throw error
+      }
+    })
+    .then(r => r.text())
+    .then((filename) => {
+      this.setState({
+      [id]: Object.assign({}, this.state[id], {'uploading': false})
+      });
+      dispatch(actions.change(`${model}.documents.${id}`, filename))
+    })
+    .catch((error) => {
+      this.setState({
+        [id]: void 0,
+        errors: {[id]: true}})
+    });
+  }
+
+  onReset(id, e) {
+    e.preventDefault();
+    const { dispatch, model } = this.props;
+    dispatch(actions.omit(`${model}.documents`, id))
+    this.setState({
+      [id]: Object.assign({}, this.state[id], {'file': void 0})
+    })
+  }
+
+  onChange(id, e) {
+    e.preventDefault();
+    this.setState({
+      [id]: Object.assign({}, this.state[id], {'file': e.target.files[0]}),
+      errors: {[id]: void 0}
+    })
+  }
+
+  render() {
+    const { action, csrf_token, model, mode, form, documentsForm, serverRender, onSubmit, pathname } = this.props;
+    return (
+      <Layout>
+        <header>
+          <h1>Documents</h1>
+        </header>
+        <article role="main">
+          <ErrorBox focusOnMount={true} model={model}/>
+
+          <StatefulError
+            model={`${model}.documents`}
+            id="documents"
+            messages={{
+              documents: 'Please upload all documents.'
+            }}
+          />
+          <Form model={model}
+            action={action}
+            method="post"
+            id="yourinfo"
+            component={SubmitForm}
+            valid={form.valid}
+            onCustomSubmit={onSubmit}
+            validators={{
+              documents: (documents) => Object.keys(documents).length === 4
+            }}
+          >
+            {
+              csrf_token && (
+              <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />
+            )}
+
+            {this.files.map((file, i) => {
```

Comment:
read this as an odd naming convention. Interpreted `this.files` as an existing object, but here it is not uploaded yet.